floyd/client/files.py

```diff
@@ -6,70 +6,79 @@
+            # Reset dirs to avoid going further down this directory.
+            # Then continue to the next iteration of os.walk, which causes
+            # everything in this directory to be ignored.
             dirs[:] = []
             floyd_logger.debug("Ignoring directory : {}".format(root))
             continue
 
         for file_name in files:
-            ignore_file = False
-            normalized_path = normalize_path(path, os.path.join(root, file_name))
-            for item in ignore_list_expanded:
-                if PurePath(normalized_path).match(item):
-                    ignore_file = True
-                    break
-
-            if ignore_file:
-                floyd_logger.debug("Ignoring file : {}".format(normalized_path))
+            file_path = unix_stle_path(os.path.join(root, file_name))
+            if ignore_path(file_path, ignore_list, whitelist):
+                floyd_logger.debug("Ignoring file : {}".format(file_name))
                 continue
 
-            file_relative_path = os.path.join(root, file_name)
-            if separator != '/':  # convert relative paths to Unix style
-                file_relative_path = file_relative_path.replace(os.path.sep, '/')
-            file_full_path = os.path.join(os.getcwd(), root, file_name)
+            unignored_files.append(os.path.join(root, file_name))
 
-            local_files.append((file_type, (file_relative_path, open(file_full_path, 'rb'), 'text/plain')))
-            total_file_size += os.path.getsize(file_full_path)
+    return unignored_files
 
-    return (local_files, sizeof_fmt(total_file_size))
+def ignore_path(path, ignore_list=[], whitelist=[]):
+    """
+    Returns a boolean indicating if a path should be ignored given an
+    ignore_list and a whitelist of glob patterns.
+    """
+    return matches_glob_list(path, ignore_list) and not matches_glob_list(path, whitelist)
 
+def matches_glob_list(path, glob_list):
+    """
+    Given a list of glob patterns, returns a boolean
+    indicating if a path matches any glob in the list
+    """
+    for glob in glob_list:
+        try:
+            if PurePath(path).match(glob):
+                return True
+        except TypeError:
+            pass
+    return False
 
-def normalize_path(project_root, path):
+def get_files_in_directory(path, file_type):
     """
-    Convert `path` to a UNIX style path, where `project_root` becomes the root
-    of an imaginery file system (i.e. becomes just an initial "/").
+    Gets the list of files in the directory and subdirectories
+    Respects .floydignore file if present
     """
-    if os.path.sep != '/':
-        path = path.replace(os.path.sep, '/')
-        project_root = project_root.replace(os.path.sep, '/')
+    local_files = []
+    total_file_size = 0
+
+    ignore_list, whitelist = FloydIgnoreManager.get_lists()
 
-    path = path[len(project_root):] if path.startswith(project_root) else path
```

Comment:
I don't think this part of the `normalize_path` method was actually necessary. The glob matching functionality of `PurePath.match()` is actually pretty sophisticated, and can deal with the relative paths we pass from the `os.walk()` variables. I wonder if we went with this approach because the examples in the documentation doesn't show examples like the ones we use?

I remember determining that this wasn't necessary when doing this refactor. But I want to think on it again before we merge, perhaps there is a case when this would be necessary (like if `get_unignored_file_paths` were called with something besides `'.'`, which is the only way we're calling it at the moment).